apps/reference/docs/guides/platform/logs.mdx

```diff
@@ -4,78 +4,177 @@ title: Logging
+### Best Practices
 
-:::note
+1. Have a WHERE statement filtering over **timestamp**
 
-In the Log Explorer, timestamps are rendered as unix microsecond timestamps. SQL queries, however, should always use the `TIMESTAMP` data type. If you are using a unix timestamp value in a query, cast the value to a `TIMESTAMP` data type.
+  Querying across your entire log retention range might seem appealing, especially when you just need to get your query results fast. However, for **Enterprise** customers that have a large retention range, you would run the risk of your queries timing out due to the length cruch time required to scan and analyze across your large dataset. We recommend optimizing queries based on timestamp filtering, to ensure
 
-:::
 
-## Templates
+2. Avoid selecting large nested objects. Select individual values instead.
+
+  When querying large object, the columnar storage engine (in this case BigQuery) will select each column associated with each nested key, resulting in a large number of columns being selected. This inadvertently impacts the query speed and may result in timeouts or memory errors, especially for projects with large amount of logging activity. 
+
+  Instead, try to select only the values required, and where this is not possible, drill down to the minimally required object keys.
+
+  ```sql
+  -- ❌ edge_logs metadata contains many nested keys
+  SELECT timestamp, m as metadata
+  FROM edge_logs t
+  CROSS JOIN UNNEST(t.metadata) as m;
+  
+  -- ✅ select only the required values
+  SELECT timestamp, r.method
+  FROM edge_logs t
+  CROSS JOIN UNNEST(t.metadata) as m
+  CROSS JOIN UNNEST(m.request) as r
+  ```
+
+### Examples and Templates
+
+For example, you may enter the following into the SQL editor to query for each user's IP address:
+
+```sql
+SELECT DATETIME(timestamp), h.x_real_ip
+FROM edge_logs
+  LEFT JOIN UNNEST(metadata) as m ON TRUE
+  LEFT JOIN UNNEST(m.request) AS r ON TRUE
+  LEFT JOIN UNNEST(r.headers) AS h ON TRUE
+WHERE h.x_real_ip IS NOT NULL AND r.method = "GET"
+```
+
+Within the Logs Explorer, we have also included some templates to help you get started on each. Templates are available under the **Templates** tab, or under the **Templates** Dropdown in the **Query** tab.
+
+## Limitations
 
-Templates are available to help craft you log queries. Templates are available under the **Templates** tab, or under the **Templates** Dropdown in the **Query** tab.
+- Logging UI is currently not available for self-hosted and local development. This is currently on the roadmap and you can follow the progress in the [Logflare repository](https://github.com/Logflare/logflare).
+- Unoptimized queries for projects with high logging activity may lead to memory errors or timeouts. Please refer to the [best practices](#best-practices) when crafting your queries.
```

Comment:
Hmm, I feel like this section is a bit redundant. Suggest moving the first bullet point to the callout at the top of the doc. The second bullet point is unnecessary—assuming the user has made it to this point in the doc, they've likely seen the best practices section.

Reply:
👍